### PR TITLE
Display indices in sections seperated by tier

### PR DESCRIPTION
**Pull request by patrickmann**
Resolves Graylog2/graylog-plugin-enterprise#8400

Extends the `GET /system/indexer/overview/${indexSetId}` API to provide tier type (`hot` or `warm`) and shard count for each index.

```
{
    "deflector": {
...
    },
    "indexer_cluster": {
...
    },
    "counts": {
        "events": 0
    },
    "indices": [
        {
            "index_name": "graylog_60",
            "size": {
                "events": 0,
                "deleted": 0,
                "bytes": 624
            },
            "range": {
                "index_name": "graylog_60",
                "begin": "1970-01-01T00:00:00.000Z",
                "end": "1970-01-01T00:00:00.000Z",
                "calculated_at": "2024-09-02T10:42:31.474Z",
                "took_ms": 0
            },
            "tier": "hot",
            "shard_count": 3,
            "is_reopened": false,
            "is_deflector": true,
            "is_closed": false
        }
    ]
}

```

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexerOverviewResource.java [hunk 1]
```diff
@@ -169,6 +168,7 @@ private IndexSummary buildIndexSummary(Map.Entry<String, JsonNode> indexStats,
         final long deleted = docs.path("deleted").asLong();
         final JsonNode store = primaries.path("store");
         final long sizeInBytes = store.path("size_in_bytes").asLong();
+        final long shardCount = indicesAdapter.getShardsInfo(indexName).size();
```
Comment (todvora) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexerOverviewResource.java:171]: It's a long here and Long in the DTO. Is there any valid situation where the value is null and we need the boxed object?

**Review by todvora — APPROVED** [review]
Generally OK, just two small details that could make the code more readable and robust.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexerOverviewResource.java (hunk 1 above)
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexerOverviewResource.java:171]: I made it @Nullable just to be on the safe side for back-compat.

Comment (todvora) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexerOverviewResource.java:171]: What would then be a situation which leads to a null value here, in a backwards-compatible case?

Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/rest/resources/system/indexer/IndexerOverviewResource.java:171]: If it were stored in the DB and you upgrade. 
Looking more closely, we never store this object in the DB. I removed the @Nullable and changed the type.

File: graylog2-server/src/main/java/org/graylog2/rest/models/system/indexer/responses/IndexSummary.java [hunk 2]
```diff
@@ -28,6 +28,8 @@
 @WithBeanGetter
 @JsonAutoDetect
 public abstract class IndexSummary {
+    public static final String HOT_TIER = "hot";
```
[the remark this answers is not in the document]
Comment (patrickmann) [reply graylog2-server/src/main/java/org/graylog2/rest/models/system/indexer/responses/IndexSummary.java]: done

**Pull request merged**